From 62d2b87b30af0dedb64840cccb9afed22f4a98d7 Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Wed, 6 Aug 2014 03:45:03 -0400
Subject: [x86] pci: Use x86_pci_root_bus_node() instead of get_mp_bus_to_node()

Message-id: <20140806034503.1438.34636.stgit@gir.stowe>
Patchwork-id: 87418
O-Subject: [RHEL7.1 PATCH 09/69] x86/PCI: Use x86_pci_root_bus_node() instead of get_mp_bus_to_node()
Bugzilla: 1110896
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Dean Nelson <dnelson@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1110896
Upstream Status: 6616dbdf6d967c704830586d5ead35c9767a8431

commit 6616dbdf6d967c704830586d5ead35c9767a8431
Author: Bjorn Helgaas <bhelgaas@google.com>
Date:   Fri Jan 24 11:54:51 2014 -0700

    x86/PCI: Use x86_pci_root_bus_node() instead of get_mp_bus_to_node()

    This replaces all uses of get_mp_bus_to_node() with x86_pci_root_bus_node().

    I think these uses are all on root buses, except possibly for blind
    probing, where NUMA node information is unimportant.

    Signed-off-by: Bjorn Helgaas <bhelgaas@google.com>
    Acked-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/pci/acpi.c b/arch/x86/pci/acpi.c
index 7fb24e5..cdd2b76 100644
--- a/arch/x86/pci/acpi.c
+++ b/arch/x86/pci/acpi.c
@@ -503,7 +503,7 @@ struct pci_bus *pci_acpi_scan_root(struct acpi_pci_root *root)
   set_mp_bus_to_node(busnum, node);
  else
 #endif
-  node = get_mp_bus_to_node(busnum);
+  node = x86_pci_root_bus_node(busnum);
 
  if (node != -1 && !node_online(node))
   node = -1;
diff --git a/arch/x86/pci/common.c b/arch/x86/pci/common.c
index 600e592..f7d6800 100644
--- a/arch/x86/pci/common.c
+++ b/arch/x86/pci/common.c
@@ -467,7 +467,7 @@ void pcibios_scan_root(int busnum)
   printk(KERN_ERR "PCI: OOM, skipping PCI bus %02x\n", busnum);
   return;
  }
- sd->node = get_mp_bus_to_node(busnum);
+ sd->node = x86_pci_root_bus_node(busnum);
  x86_pci_root_bus_resources(busnum, &resources);
  printk(KERN_DEBUG "PCI: Probing PCI hardware (bus %02x)\n", busnum);
  bus = pci_scan_root_bus(NULL, busnum, &pci_root_ops, sd, &resources);
-- 
1.7.1